### PR TITLE
Foundation: Request/Response Pipeline Refactoring

### PR DESCRIPTION
**Pull request by krcpa**
## Summary

This PR implements the foundational request/response pipeline refactoring as outlined in issue #239, completing task 3 of 12 in the HTTP layer refactoring epic (#236).

### Key Changes:

• **RequestPipeline Integration**: Created unified request processing pipeline that integrates seamlessly with the new framework-native routing engine
• **Enhanced Parameter Extraction**: Added typed parameter extraction methods for ElifRequest with proper error handling and validation
• **Response Helper Methods**: Extended ElifResponse with comprehensive helper methods for common response patterns
• **Framework Independence**: All functionality works with pure elif types, no Axum exposure

#### RequestPipeline Features:
- Route resolution using the new RouteMatcher
- Parameter injection from route matches
- Middleware pipeline execution with route-specific configurations
- Comprehensive error handling with `PipelineError` types
- Builder pattern for easy configuration

#### Enhanced ElifRequest:
- Typed parameter extraction: `path_param_typed<T>()`, `query_param_typed<T>()`
- Convenience methods for common types: `path_param_int()`, `path_param_uuid()`
- Validation helpers: `validate_path_param()`, parameter existence checks
- Consistent error handling with `ParamError`

#### ElifResponse Improvements:
- File serving with automatic MIME type detection
- Enhanced response builders: JSON, XML, CSV, JavaScript, CSS
- CORS support and security headers
- Caching headers with ETag generation
- Image and binary response helpers
- Response transformation utilities

### Test Coverage:
- 314 tests passing
- Comprehensive test coverage for all new functionality
- Integration tests for pipeline processing
- Parameter extraction validation tests

## Test plan

- [x] All existing tests continue to pass
- [x] New pipeline tests validate core functionality
- [x] Parameter extraction tests cover edge cases
- [x] Response helper tests ensure proper behavior
- [x] Build verification successful

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: crates/elif-http/src/request/pipeline.rs [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+        
+        // 2. Parameter injection
+        let request_with_params = self.inject_params(request, &route_match);
+        
+        // 3. Build complete middleware pipeline for this route
+        let complete_pipeline = self.build_route_pipeline(&route_match)?;
+        
+        // 4. Execute middleware + handler
+        let route_id = route_match.route_id.clone();
+        let handlers = self.handlers.clone();
```
[the remark this answers is not in the document]
Comment (krcpa) [reply crates/elif-http/src/request/pipeline.rs]: @gemini-code-assist check please.

File: crates/elif-http/src/request/pipeline.rs [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+        let http_method = match request.method.to_axum() {
+            &axum::http::Method::GET => HttpMethod::GET,
+            &axum::http::Method::POST => HttpMethod::POST,
+            &axum::http::Method::PUT => HttpMethod::PUT,
+            &axum::http::Method::DELETE => HttpMethod::DELETE,
+            &axum::http::Method::PATCH => HttpMethod::PATCH,
+            &axum::http::Method::HEAD => HttpMethod::HEAD,
+            &axum::http::Method::OPTIONS => HttpMethod::OPTIONS,
+            &axum::http::Method::TRACE => HttpMethod::TRACE,
+            _ => HttpMethod::GET, // Fallback for any unknown method
```
[the remark this answers is not in the document]
Comment (krcpa) [reply crates/elif-http/src/request/pipeline.rs:168]: @gemini-code-assist check please.

[the remark this answers is not in the document]
Comment (krcpa) [reply crates/elif-http/src/request/pipeline.rs:168]: @gemini-code-assist check if this is fixed.

File: crates/elif-http/src/response/response.rs [hunk 3]
```diff
@@ -477,6 +477,325 @@ impl ElifResponse {
...
+        // Hash header keys and values for ETag generation
+        for (key, value) in self.headers.iter() {
+            key.as_str().hash(&mut hasher);
+            if let Ok(value_str) = value.to_str() {
+                value_str.hash(&mut hasher);
+            }
+        }
+        
+        format!("{:x}", hasher.finish())
+    }
```
[the remark this answers is not in the document]
Comment (krcpa) [reply crates/elif-http/src/response/response.rs]: @gemini-code-assist check if this is fixed.
